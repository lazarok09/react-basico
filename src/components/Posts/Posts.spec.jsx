import { render, screen } from "@testing-library/react";
import { Posts } from "./index.jsx";

const props = {
  posts: [
    {
      id: 1,
      title: "title example",
      cover: "img/img.png",
      body: "paragraph example",
    },
    {
      id: 2,
      title: "title example2",
      cover: "img/img2.png",
      body: "paragraph example2",
    },
    {
      id: 3,
      title: "title example3",
      cover: "img/img3.png",
      body: "paragraph example3",
    },
  ],
};

describe("<Posts />", () => {
  it("should render posts", () => {
    render(<Posts {...props} />);
    // verifica o nome dentro dos heading
    expect(
      screen.getAllByRole("heading", { name: /title example/i })
    ).toHaveLength(3);
    // verificar o alt text da imagem
    expect(screen.getAllByRole("img", { name: /title example/i })).toHaveLength(
      3
    );
    // verificar os textos
    expect(screen.getAllByText(/paragraph example/i)).toHaveLength(3);
    // verifica se uma imagem com o titulo x e o atributo de endereço src informado existe
    expect(screen.getByRole("img", { name: "title example" })).toHaveAttribute(
      "src",
      "img/img.png"
    );
  });
  // se não tiver props no post
  it("should not render posts if there's no props />", () => {
    render(<Posts />);

    expect(
      screen.queryByRole("heading", { name: /title example/ })
    ).not.toBeInTheDocument();
  });
  // snapshot
  it("should match snapshot", () => {
    const { container } = render(<Posts {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
