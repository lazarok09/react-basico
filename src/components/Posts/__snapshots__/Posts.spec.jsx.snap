// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> should match snapshot 1`] = `
<div>
  <div
    class="post-card"
  >
    <img
      alt="title example"
      src="img/img.png"
    />
    <h2>
      title example
    </h2>
    <p>
      paragraph example
    </p>
  </div>
  <div
    class="post-card"
  >
    <img
      alt="title example2"
      src="img/img2.png"
    />
    <h2>
      title example2
    </h2>
    <p>
      paragraph example2
    </p>
  </div>
  <div
    class="post-card"
  >
    <img
      alt="title example3"
      src="img/img3.png"
    />
    <h2>
      title example3
    </h2>
    <p>
      paragraph example3
    </p>
  </div>
</div>
`;
